datalad/distribution/tests/test_create_sibling.py

```diff
@@ -506,6 +507,9 @@ def check_failon_no_permissions(use_ssh, src_path, target_path):
 
 def test_failon_no_permissions():
     skip_if_on_windows()
+    if os.environ.get("NOSE_WRAPPER", "").startswith("sudo"):
+        raise SkipTest(
+            "Skipping 'no permissions' check because running under sudo")
```

Comment:
packages which run tests at build time could be built with root permissions.
For that purpose we already have similar skips:
```
$> git grep skip.*sudo
datalad/support/tests/test_annexrepo.py:@skip_if(cond=(on_windows or os.geteuid() == 0))  # uid and sudo not available on windows
datalad/support/tests/test_annexrepo.py:@skip_if(cond=(on_windows or os.geteuid() == 0))  # uid and sudo not available on windows
```
might be worth generalizing it into `skip_root_user` or alike to avoid duplicates at this point

Reply:
>  For that purpose we already have similar skips:

Thanks for pointing that out.  I've added a helper.

<details>
<summary>range-diff</summary>

```
-:  --------- > 1:  75505b721 TST: utils: Add skip_if_root helper
1:  01a5cf732 ! 2:  0993fcbb6 TST: create-sibling: Skip permissions test when running under sudo
    @@ datalad/distribution/tests/test_create_sibling.py
          ok_file_under_git,
          skip_if_on_windows,
          skip_ssh,
    -+    SkipTest,
    ++    skip_if_root,
          slow,
          swallow_logs,
          with_tempfile,
    -@@ datalad/distribution/tests/test_create_sibling.py: def check_failon_no_permissions(use_ssh, src_path, target_path):
    +@@ datalad/distribution/tests/test_create_sibling.py: def test_target_ssh_since():
    +     yield check_target_ssh_since, False
      
    - def test_failon_no_permissions():
    -     skip_if_on_windows()
    -+    if os.environ.get("NOSE_WRAPPER", "").startswith("sudo"):
    -+        raise SkipTest(
    -+            "Skipping 'no permissions' check because running under sudo")
    -     yield skip_ssh(check_failon_no_permissions), True
    -     yield check_failon_no_permissions, False
      
    ++@skip_if_root
    + @with_tempfile(mkdir=True)
    + @with_tempfile(mkdir=True)
    + def check_failon_no_permissions(use_ssh, src_path, target_path):
-:  --------- > 3:  b6aa5fb7b TST: create-sibling: Reposition skip_if_on_windows call
```

</details>

I've added back the temporary tip commit to test the `sudo -E` cron run.  I'll drop that if things look good.

Reply:
> I've added back the temporary tip commit to test the `sudo -E` cron run. I'll drop that if things look good.

All green: https://travis-ci.org/github/datalad/datalad/builds/687477156